drivers/flash/Kconfig.stm32

```diff
@@ -13,6 +14,27 @@ config SOC_FLASH_STM32
 	default y
 	select FLASH_PAGE_LAYOUT
 	select FLASH_HAS_PAGE_LAYOUT
+	select FLASH_HAS_EX_OP
 	select MPU_ALLOW_FLASH_WRITE if ARM_MPU
 	help
 	  Enable flash driver for STM32 series
+
+if SOC_FLASH_STM32
+
+config FLASH_STM32_WRITE_PROTECT
+	bool "Extended operation for flash write protection control"
+	depends on SOC_SERIES_STM32F4X
+	default n
+	help
+	  Enables flash extended operation for enabling/disabling flash write
+	  protection.
+
+config FLASH_STM32_BLOCK_WRITE_PROTECT_DISABLING
```

Comment:
Try naming Kconfigs in some structured way, for example here you have `FLASH_STM32_WRITE_PROTECT`  and the `FLASH_STM32_BLOCK_WRITE_PROTECT_DISABLING` is additional option to `FLASH_STM32_WRITE_PROTECT`. `FLASH_STM32_WRITE_PROTECT_DISABLE_PREVENTION` (sorry could not get myself to write BLOCK), would be more intuitive.